reporter/src/main.rs

```diff
@@ -123,6 +109,37 @@ fn usage() -> ! {
     std::process::exit(1)
 }
 
+fn geomean(vs: &[f64]) -> f64 {
+    let prod = vs.iter().fold(1.0, |p, v| p * v);
+    prod.powf(1.0 / vs.len() as f64)
+}
+
+/// Compute the minimum value of 64-bit floats.
+///
+/// Panics if the comparison is invalid.
+fn f64_min(vs: &[f64]) -> f64 {
+    *vs.iter().min_by(|a, b| a.partial_cmp(b).unwrap()).unwrap()
```

Comment:
Can this not just be `vs.iter().min()`? I don't think you need a helper function or the `partial_cmp` stuff.

Reply:
You have to do this because not all values in the float lattice are comparable.

Reply:
I don't get it.

Reply:
Hang on, now I'm confused. If that were true, then `some_float < other_float` wouldn't return bool...

All I can tell you is, you can't do that for some reason:
```
   Compiling playground v0.0.1 (/playground)
error[E0277]: the trait bound `{float}: Ord` is not satisfied
    --> src/main.rs:3:38
     |
3    |     println!("{}", [1.0, 2.0].iter().min())
     |                                      ^^^ the trait `Ord` is not implemented for `{float}`, which is required by `&{float}: Ord`
     |
     = help: the following other types implement trait `Ord`:
               i128
               i16
               i32
               i64
               i8
               isize
               u128
               u16
             and 4 others
     = note: required for `&{float}` to implement `Ord`
```

Reply:
The error makes sense, floats are not totally ordered, but then why does `<=` work? :P